ui/src/components/RecordButton.js

```diff
@@ -105,20 +105,26 @@ export class RecordButton extends LitElement {
                 return;
             }
 
-            // Attempt to upload it
-            if (audio.recordingUrl) {
-                try {
-                    const resp = await this.utteranceService.saveAudio(audio);
+            // Attempt to upload the audio file
+            this.uploadAudio(audio)
+        }
+    }
 
-                    if (!resp) throw new Error();
-                } catch (e) {
-                    // If upload failed, pivot to upload error collection state
-                    this.dispatchCollectionState(CollectionStates.UPLOAD_ERROR);
-                }
-            }
+    /**
+     * Attempts to upload the current audio file to the backend.
+     * @param {Object} audio An object containing an audio Blob and its 
+     *  corresponding URL.
+     */
+    async uploadAudio(audio) {
+        if (audio.recordingUrl) {
+            this.handleSaveAudio(audio);
+            const response = await this.utteranceService.saveAudio(audio);
 
-            this.handleFinish();
+            if (!response) {
+                dispatchErrorToast(this, `Audio failed to upload. Retry?`, true);
```

Comment:
nit, create function `dispatchRetryToast` which sets the retry param

Reply:
created